### ISSUE TITLE
New version of zennit produces wrong LRP heatmaps

### ISSUE THREAD

**Issue by rachtibat** [chr5tphr/zennit#44]
Hi,

it is a long time ago, I pulled all the new zennit updates. So I missed a lot of comments and I can not exactly point out where the LRP pipeline seems to have broken down the first time.
For test purposes, I always use the same LeNet model pretrained on FashionMNIST.
When I generate a heatmap with "epsilon_plus", I get with the old version of zennit the same heatmap as in Innvestigate. But the new version suddenly produces completely different heatmaps.

You can try it out with my code example on https://github.com/rachtibat/mnist_zennit I uploaded for you.
Just change the import "zennit_old" to "zennit_new" and you see the difference. I changed the name of zennit.torchvision to zennit.torchvision_2 making it work fast and dirty (; (see https://github.com/chr5tphr/zennit/pull/6)

Do you think, I made a mistake or is zennit somehow computing the heatmaps differently?

Best

P.s. if this is really a bug, it might solve the other issues

**Comment by rachtibat** [comment]
Using "jet" cmap
Old zennit:
![old](https://user-images.githubusercontent.com/48683081/126989372-b9c2bafe-9d75-4beb-9df9-03903008ef79.png)
![old 1](https://user-images.githubusercontent.com/48683081/126989399-17cfa225-f4fa-4e41-a053-aabe05f49f0c.png)

New Zennit:
![new](https://user-images.githubusercontent.com/48683081/126989461-ce6d70e8-5ecb-4ce4-ad35-04e6201b8941.png)
![new 1](https://user-images.githubusercontent.com/48683081/126989489-d665c032-9b3b-4d59-a684-04fa637bf846.png)

**Comment by chr5tphr** [comment]
I'll have a look.
Two things first:
- in [line 46 of main.py](https://github.com/rachtibat/mnist_zennit/blob/master/main.py#L46), the max should be absolute
- Why not use the zennit package directly installed from PyPI or github? If you do not like virtual environments, you can also use [`__pypackages__`](https://www.python.org/dev/peps/pep-0582/). This makes it much easier to update and keep track of versions.

**Comment by chr5tphr** [comment]
Found the problem, the gradient check in [c984127](https://github.com/chr5tphr/zennit/commit/c9841270ba5b7f6a0a3442d986cb96ef79417a57) was only checking for `.grad_fn`. In the first layer, there may not be a `.grad_fn` yet, thus it should rather check for `.requires_grad`. I will prepare a MR.

**Comment by rachtibat** [comment]
Hi,
Thanks for your suggestions.
I tested your fix and it's working great!

Best
